### ISSUE TITLE
Directory with invalid characters

### ISSUE THREAD

**Issue by Craftplacer** [maharmstone/btrfs#337]
When creating a folder in Visual Studio, it manages to create a directory with a trailing backslash which hasn't been sanitized.
(Theory being it creates a folder called "_`[name]`_`\`", in another one, that being the solution folder in this case)

It becomes undeletable because of Windows limitations, scrubbing hasn't fixed it.

##### reproduction steps:
1. create a new solution in Visual Studio
2. create a new directory under the newly created project of the solution
3. let Visual Studio fail to create a directory with invalid characters

##### screenshot:
![image](https://user-images.githubusercontent.com/22963120/103731395-56f60180-4fe5-11eb-9854-f911b41f91ff.png)

##### error log from Visual Studio after creating the directory:
```
=====================
06.01.2021 05:52:36
Recoverable
System.ArgumentException: Illegal characters in path.
   at System.Security.Permissions.FileIOPermission.EmulateFileIOPermissionChecks(String fullPath)
   at System.Security.Permissions.FileIOPermission.QuickDemand(FileIOPermissionAccess access, String fullPath, Boolean checkForDuplicates, Boolean needFullPath)
   at Microsoft.VisualStudio.PlatformUI.Packages.FileChange.DefaultFileSystem.PathWrapper.GetFullPath(String path)
   at Microsoft.VisualStudio.Services.FileChangeService.ValidateAndNormalize(String path, String paramName)
   at Microsoft.VisualStudio.Services.FileChangeService.AdviseFileChangeCore(String filename, _VSFILECHANGEFLAGS filter, EventSinkBundle bundle, CancellationToken token)
   at Microsoft.VisualStudio.Services.FileChangeService.<>c__DisplayClass89_0.<AdviseFileChangeAsync>b__0(CancellationToken token)
   at Microsoft.VisualStudio.Services.FileChangeService.<RunWithTracingAsync>d__125`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.ProjectEvaluationAdditionalDependentFileWatchingService.<CheckContentIrrelevantFileDependenciesStateAsync>d__22.MoveNext()
===================
```

**Comment by maharmstone** [comment]
Thanks. This looks to be the same issue as https://github.com/maharmstone/btrfs/issues/334 - it's fixed in the source, but not yet in a released version.

**Comment by maharmstone** [comment]
Fixed in v1.7.6
